apps/opik-backend/src/main/java/com/comet/opik/domain/llmproviders/LlmProviderAnthropicMapper.java

```diff
@@ -0,0 +1,122 @@
+    AssistantMessage toAssistantMessage(@NonNull AnthropicContent content);
+
+    @Mapping(expression = "java(request.model())", target = "model")
+    @Mapping(expression = "java(request.stream())", target = "stream")
+    @Mapping(expression = "java(request.temperature())", target = "temperature")
+    @Mapping(expression = "java(request.topP())", target = "topP")
+    @Mapping(expression = "java(request.stop())", target = "stopSequences")
+    @Mapping(expression = "java(request.maxCompletionTokens())", target = "maxTokens")
+    @Mapping(source = "request", target = "messages", qualifiedByName = "mapToMessages")
+    @Mapping(source = "request", target = "system", qualifiedByName = "mapToSystemMessages")
+    AnthropicCreateMessageRequest toCreateMessageRequest(@NonNull ChatCompletionRequest request);
+
+    @Named("mapToChoices")
+    default List<ChatCompletionChoice> mapToChoices(@NonNull AnthropicCreateMessageResponse response) {
+        if (response.content == null || response.content.isEmpty()) {
+            return List.of();
+        }
+        return response.content.stream().map(content -> toChoice(content, response)).toList();
+    }
+
+    @Named("mapToUsage")
+    default Usage mapToUsage(AnthropicUsage usage) {
+        if (usage == null) {
+            return null;
+        }
+
+        return Usage.builder()
+                .promptTokens(usage.inputTokens)
+                .completionTokens(usage.outputTokens)
+                .totalTokens(usage.inputTokens + usage.outputTokens)
+                .build();
+    }
+
+    @Named("mapToMessages")
+    default List<AnthropicMessage> mapToMessages(@NonNull ChatCompletionRequest request) {
+        return request.messages().stream()
+                .filter(message -> List.of(Role.ASSISTANT, Role.USER).contains(message.role()))
+                .map(this::mapToAnthropicMessage).toList();
+    }
+
+    @Named("mapToSystemMessages")
+    default List<AnthropicTextContent> mapToSystemMessages(@NonNull ChatCompletionRequest request) {
+        return request.messages().stream()
+                .filter(message -> message.role() == Role.SYSTEM)
+                .map(this::mapToSystemMessage).toList();
+    }
+
+    default AnthropicMessage mapToAnthropicMessage(@NonNull Message message) {
+        if (message instanceof AssistantMessage assistantMessage) {
+            return AnthropicMessage.builder()
+                    .role(AnthropicRole.ASSISTANT)
+                    .content(List.of(new AnthropicTextContent(assistantMessage.content())))
+                    .build();
+        }
+
+        if (message instanceof UserMessage userMessage) {
+            return AnthropicMessage.builder()
+                    .role(AnthropicRole.USER)
+                    .content(List.of(toAnthropicMessageContent(userMessage.content())))
+                    .build();
+        }
+
+        throw new BadRequestException("unexpected message role: " + message.role());
```

Comment:
The switch case would be more concise here.

Reply:
Done